tests_tf/test_attacks.py

```diff
@@ -255,7 +255,7 @@ def test_generate_np_can_be_called_with_different_eps(self):
                                       clip_min=-5.0, clip_max=5.0)
 
       delta = np.max(np.abs(x_adv - x_val), axis=1)
-      self.assertClose(delta, eps)
+      self.assertTrue(np.all(delta<=(eps+1e-4)))
```

Comment:
In the generic version where the constraint might not be tight, you should use assertLessEqual rather than assertTrue.